### DISCUSSION TITLE
Model events: how to access posted data?

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#622] [Q&A (Help)] [answered]
I need to intercept a POST when creating a record to manipulate data .
I can access $item, ok.
But how can I access raw field data?

**Comment by promatik** [comment, marked as answer]
Hey @realtebo!
I'm not sure in which part of the code you are, but I believe you can use the `request()` at any place.
So most probably, if you want to intercept a request to manipulate data you can use the event `saving` (that gets called before any save, create/update) and there you can access the `request()` data.

Also, if you just want to manipulate data on your model before save, most probably you need a mutator.
Take a look here; https://laravel.com/docs/10.x/eloquent-mutators#defining-a-mutator

**Reply by pxpm** [reply]
Just o add up that Backpack also provides an helper to transform the request when it comes from a Backpack Form posted with AJAX. 
 https://github.com/Laravel-Backpack/CRUD/blob/0e8cda5e5ea5e4e503a5a967fd617b9bdaa4118e/src/helpers.php#L58C26-L58C26

Notice that Backpack will send the form value inside the `form` key when posted with ajax.

**Comment by realtebo** [comment]
I am creating model events /saving and saved/ inside setupCreateOperation

I resolved using request()
